libs/wingsdk/src/shared-aws/function.inflight.ts

```diff
@@ -25,14 +129,66 @@ export class FunctionClient implements IFunctionClient {
       );
     }
     if (!response.Payload) {
-      return "";
+      return { payload: "" };
     }
     const value = JSON.parse(toUtf8(response.Payload)) ?? "";
-    if (typeof value !== "string") {
-      throw new Error(
-        `function returned value of type ${typeof value}, not string`
+    return value;
+  }
+
+  /**
+   * Invoke the function, passing the given payload as an argument.
+   *  @returns the function returned payload only
+   */
+  public async invoke(payload: string): Promise<string> {
+    const arnParts = this.functionArn.split(":");
+    const functionName = arnParts[arnParts.indexOf("function") + 1];
+    const value = await this.executeFunction(payload);
+
+    // kind of hacky, but this is the most convenient way to pass those arguments to the calling function
+    console.log(
+      `Invoking ${functionName}:\t${Buffer.from(
+        JSON.stringify({
+          logGroupName: value?.context?.logGroupName,
+          logStreamName: value?.context?.logStreamName,
+          constructPath: this.constructPath,
+        }),
+        "binary"
+      ).toString("base64")}`
+    );
+
+    return this.verify(value);
+  }
+
+  /**
+   * Invoke the function, passing the given payload as an argument.
+   *
+   * @returns the function returned payload and logs
+   */
+  public async invokeWithLogs(
+    payload: string,
+    logType: LogType = LogType.DEFAULT
+  ): Promise<[string, Trace[]]> {
+    const traces: Trace[] = [];
+
+    const value = await this.executeFunction(payload);
+
+    if (
+      value?.context?.logGroupName &&
+      value?.context?.logStreamName &&
+      logType !== LogType.NONE
+    ) {
+      // waiting for the logs to be created
+      await new Promise((resolve) => setTimeout(resolve, 10000));
```

Comment:
Does this mean every time a user runs `wing test` on the cloud they will need to wait an extra ten seconds to get feedback of their test results? We should probably avoid making developers wait longer to get feedback in "happy path" situations. (ten seconds is a lot!)

I'd think of the logs shown in `wing test` as a *best effort* mechanism. A function can produce thousands of logs, so what's most important is handling common use cases well (like showing the last several dozens of logs that were printed). When that isn't enough, the user should be provided enough information (like the name of the function and the request ID) to continue their debugging. Maybe for tools like the Wing Console we will add some other kind of moethod like `tailLogs()` for when you want to run a test and stream all of the logs as the test runs. But I feel like we should treat that as a separate use case since it requires extra resources. WDYT?

Reply:
changed the mechanism a bit now to prevent waiting, and logging extra chars,
I like the `tailLogs` idea, we can add it to the function execution in the future, should I open an issue?